pkgs/applications/networking/irc/weechat/wrapper.nix

```diff
@@ -0,0 +1,29 @@
+let
+  weechatDefault = "$HOME/.weechat";
+in
+  { weechat, plugins ? {}, buildFHSUserEnv
+  , homeDir ? weechatDefault, term ? "xterm" }:
+
+  let
+    weechat' =
+      if homeDir == weechatDefault
+      then weechat
+      else weechat.override { inherit homeDir; };
+
+    termScript = ''
+      export TERM=${term}
+    '';
+
+    pluginScripts = map (a: ''
+      mkdir -p ${homeDir}/$(dirname ${a})
+      cp -f ${plugins."${a}"}/share/$(basename ${a}) ${homeDir}/${a}
```

Comment:
would it not be easier to just do `export $HOME=$(mktemp -d)` instead of setting up a mount namespace.